metanorm/normalizers/url.py

```diff
@@ -0,0 +1,201 @@
+        "ftp://ftp.remss.com/gmi/": 'Atmosphere parameters from Global Precipitation Measurement Microwave Imager',
+        "ftp://ftp.gportal.jaxa.jp/standard": 'AMSR2-L2 Sea Surface Temperature'}
+
+    urls_dsp = {'ftp://anon-ftp.ceda.ac.uk/neodc/esacci/sst/': ['sea_surface_temperature'],
+                'ftp://ftp.gportal.jaxa.jp/standard/GCOM-W/GCOM-W.AMSR2/L2.SST':
+                ['sea_surface_temperature'],
+                'ftp://ftp.gportal.jaxa.jp/standard/GCOM-W/GCOM-W.AMSR2/L3.SST_10':
+                ['sea_surface_temperature'],
+                'ftp://ftp.gportal.jaxa.jp/standard/GCOM-W/GCOM-W.AMSR2/L3.SST_25':
+                ['sea_surface_temperature'],
+                "ftp://ftp.remss.com/gmi/":
+                ['wind_speed', 'atmosphere_mass_content_of_water_vapor',
+                 'atmosphere_mass_content_of_cloud_liquid_water', 'rainfall_rate'], }
+
+    @staticmethod
+    def find_matching_value(associated_dict, raw_attributes):
+        """ Loop through <associated_dict> and get the matching value  """
+        if 'url' in raw_attributes:
+            for url in associated_dict.keys():
+                if raw_attributes['url'].startswith(url):
+                    return associated_dict[url]
+        return None
+
+    @staticmethod
+    def extract_time(time_text):
+        """ Extract time from file name """
+        if not time_text:
+            return None
+        # tuple of all formats for usage of "strptime" function of datetime
+        # Order of this tuple matters! so the more generic formats must be in the end of tuple
+        strp_format = (
+            "f35_%Y%m%dv8.2_d3d.gz",
+            "f35_%Y%m%dv8.2.gz",
+            "f35_%Y%mv8.2.gz",
+            # for normal daily files of jaxa GCOM-W.AMSR2 folder
+            "GW1AM2_%Y%m%d",
+            # for month files of jaxa GCOM-W.AMSR2 (MUST below the one for daily files)
+            "GW1AM2_%Y%m%H",
+            "%Y%m%d%H%M",
+            "%Y%m%d",
+        )
+        for _format in strp_format:
+            try:
+                extracted_date = datetime.strptime(time_text, _format).replace(tzinfo=tzutc())
+            except ValueError:
+                continue  # if the format is incorrect, then try another format for "strptime"
+            else:
+                return extracted_date
+
+    @staticmethod
+    def length_of_month(extracted_date):
+        """ length of the month that 'extracted_date' has been found in it """
+        return \
+            relativedelta(days=calendar.monthrange(
+                extracted_date.year, extracted_date.month)[1] - 1)
+
+    def get_platform(self, raw_attributes):
+        """ return the corresponding platfrom based on specified ftp source """
+        founded_values = self.find_matching_value(self.urls_platforms, raw_attributes,)
+        if founded_values:
+            return pti.get_gcmd_platform(founded_values)
```

Comment:
Yes, now I remember why I previously suggested to apply a desired_function inside the `find_matching_value`: it doesn't require to use `if` in every `get_` method. Now we are back to repetition. Maybe it is better readable but it is not DRY and the same mistake has to be fixed in several places. For example, `found` instead of `founded`, and `value` instead of `values`, and no comma after the last parameter in every call to `find_matching_value`.

It is up to you how to call find_matching_value but at least fix the variable name and the comma here and everywhere.
```suggestion
        found_value = self.find_matching_value(self.urls_platforms, raw_attributes)
        if found_value:
            return pti.get_gcmd_platform(found_value)
```